win32/win_fopen.h

```diff
@@ -0,0 +1,48 @@
+//
+//  Copyright(C) 2021 Roman Fomin
+//
+//  This program is free software; you can redistribute it and/or
+//  modify it under the terms of the GNU General Public License
+//  as published by the Free Software Foundation; either version 2
+//  of the License, or (at your option) any later version.
+//
+//  This program is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+//  GNU General Public License for more details.
+//
+// DESCRIPTION:
+//      unicode paths for fopen() on Windows
+
+#ifndef __WIN_D_FOPEN__
+#define __WIN_D_FOPEN__
+
+#ifdef _WIN32
+#include <stdio.h>
+#include <sys/stat.h>
+
+FILE* D_fopen(const char *filename, const char *mode);
+int D_remove(const char *path);
+int D_rename(const char *oldname, const char *newname);
+int D_stat(const char *path, struct stat *buffer);
+int D_open(const char *filename, int oflag);
+int D_access(const char *path, int mode);
+int D_mkdir(const char *dirname);
+
+#undef  fopen
```

Comment:
The whole idea is to not write explicit wrappers. I think I should close PR then.

Reply:
Even if we do end up going with transparent wrapper I think it should be in some more general include file. This way it would be harder to forget it. If some some calls are wrapped and others are not that will also result in nasty debugging sessions.

Reply:
Yes, I think making it into a wrapper is the best approach. In general it's nicer to be able to avoid scattering #ifdefs through the codebase. The other thing that's useful is that it's easy to grep through the codebase to find places where we're still using the old name. And we can easily add presubmits to check new code uses the new functions.

A good place for these functions to live would be m_misc.c where we have some similar wrappers like M_StringDuplicate and M_snprintf.

Reply:
Oh and also: at some point I'd like to merge some of the work from the c++-clean branch, where a bunch of these functions have been factored out into an explicit "safe" library that does the same thing on every system. Long term it should maybe live there, but in the short term it can go to m_misc

Reply:
Never mind, disregard message I just deleted about the c++-clean branch.